File: API/SchedHoliday/Repo/Message/DTOMessage.cs
using System.ComponentModel.DataAnnotations;
using System.Xml.Linq;
using SchedHoliday.Repo.Holiday;
using SchedHoliday.Repo.User;

namespace SchedHoliday.Repo.Message
{
    public class DTOMessage
    {
        public string Id { get; set; }

        public string Content { get; set; }

        public DateTime Date { get; set; }

        public string SenderId { get; set; } 

        public string HolidayId { get; set; }

        public DTOUser Sender { get; set; }

        public DTOHoliday Holiday { get; set; }
    }
}

File: Mobile/SchedHoliday/app/src/main/java/be/helmo/schedholiday/repository/Message/DTOMessage.kt
package be.helmo.schedholiday.repository.Message

import kotlinx.serialization.Serializable
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json

@Serializable
class DTOMessage(
    val Content:String,
    val Epoch: Long,
    var Sender : String,
    val HolidayId : String
)  {
    fun toJson() : String{
        return Json.encodeToString(this)
    }

}